lib/cli/src/frameworks/web-components/ts/Button.ts

```diff
@@ -0,0 +1,33 @@
+import { html } from 'lit-html';
+
+export interface ButtonProps {
+  primary: boolean;
+  backgroundColor: string;
+  size: string;
+  label: string;
+  onClick: () => void;
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+export const Button = ({
+  primary,
+  backgroundColor,
+  size,
+  label,
+  onClick,
+}: Partial<ButtonProps>) => {
+  const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';
+
+  return html`
+    <button
+      type="button"
+      class=${['storybook-button', `storybook-button--${size || 'medium'}`, mode].join(' ')}
+      style=${backgroundColor && { backgroundColor }}
```

Comment:
To be aligned with lit best practice we should use `ifDefined(backgroundColor)` here. It can be imported from `lit-html/directives/if-defined`

Reply:
I'm not a web component expert but if I'm right, there's other options than lit-html, we tried to limit this dependency in the code we provide. 
@shilman  WDYT ?

Reply:
Yeah I get that. I wasn't sure either, we want it to be as generic as possible so it isn't tied into one library. 

I've since realised too I wasn't quite right as in lit we wouldn't even set this property. We could just leave it as an example if you want to keep it generic though